### PR TITLE
order by primary clause

### PR DESCRIPTION
**Pull request by jiyfhust**
### What problem does this PR solve? 

https://github.com/pingcap/dumpling/issues/187

### Check List 

Tests 

For test， i add  a line to log the orderByClause.

1> MySQL version 5.7.35

 CREATE TABLE `a1` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin DEFAULT '',
  `age` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:47:21.189 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause=]

CREATE TABLE `a2` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:47:21.527 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY `id`"]

CREATE TABLE `a3` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:47:21.866 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY `name`"]

CREATE TABLE `a4` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`,`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:47:22.184 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY `id`, `name`"]

CREATE TABLE `a5` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`name`,`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:47:22.518 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY `name`, `id`"]

1> TiDB version v4.0.6

 CREATE TABLE `a1` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin DEFAULT '',
  `age` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:50:32.870 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY _tidb_rowid"]

CREATE TABLE `a2` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:50:33.158 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY `id`"]

CREATE TABLE `a3` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:50:33.442 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY _tidb_rowid"]

CREATE TABLE `a4` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`,`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:50:33.720 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY _tidb_rowid"]

CREATE TABLE `a5` (
  `id` int(11) NOT NULL,
  `name` char(10) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `age` int(11) DEFAULT NULL,
  PRIMARY KEY (`name`,`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin

[2020/11/06 17:50:34.004 +08:00] [INFO] [sql.go:209] [orderByClause] [orderByClause="ORDER BY _tidb_rowid"]

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: v4/export/sql.go [hunk 1]
```diff
@@ -294,19 +294,23 @@ func buildOrderByClause(conf *Config, db *sql.Conn, database, table string) (str
...
+        if len(cols) == 0 {
+	        return "", nil
+        }
+        odSql := "ORDER BY "
+        for i, col := range cols {
+                odSql += fmt.Sprintf("`%s`", escapeString(col));
+                if i != len(cols) - 1 {
+                        odSql += ", "
+                }
+        }
```
Comment (lichunzhu) [inline v4/export/sql.go:312]: These lines' indentation looks strange. Can you keep it the same as the previous lines'?

File: v4/export/sql.go [hunk 2]
```diff
@@ -294,19 +294,23 @@ func buildOrderByClause(conf *Config, db *sql.Conn, database, table string) (str
...
 	}
-	tableContainsPriKey := pkName != ""
-	if tableContainsPriKey {
-		return fmt.Sprintf("ORDER BY `%s`", escapeString(pkName)), nil
-	}
-	return "", nil
+        if len(cols) == 0 {
+	        return "", nil
+        }
+        odSql := "ORDER BY "
```
Comment (lichunzhu) [inline v4/export/sql.go:306]: I prefer `orderSql` or `orderBySql` here.

**Review by lichunzhu — COMMENTED** [review]
Could you please help fix the unit test? Rest LGTM
You need to mock the primary key query in `sql_test.go`.

File: v4/export/sql.go [hunk 3]
```diff
@@ -315,6 +319,26 @@ func GetColumnTypes(db *sql.Conn, fields, database, table string) ([]*sql.Column
 	return rows.ColumnTypes()
 }
 
+func GetPrimaryKeyColumns(db *sql.Conn, database, table string) ([]string, error) {
+	priColQuery := "SELECT COLUMN_NAME FROM information_schema.KEY_COLUMN_USAGE " +
+		"WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ? AND CONSTRAINT_NAME = 'PRIMARY' order by ORDINAL_POSITION;"
+	var cols []string
+	rows, err := db.QueryContext(context.Background(), priColQuery, database, table)
+        if err != nil {
+                return cols, withStack(errors.WithMessage(err, priColQuery))
```
Comment (lichunzhu) [inline v4/export/sql.go:328]: We have already updated this part. You can use `errors.Annotatef(err, "sql: %s", priColQuery)` instead.

**Comment by lichunzhu** [comment]
@kennytm PTAL

File: v4/export/sql.go (hunk 1 above)
Comment (jiyfhust) [reply v4/export/sql.go:294]: Ok, wait a moment.

File: v4/export/sql.go (hunk 3 above)
Comment (jiyfhust) [reply v4/export/sql.go]: Hi, it is all done. Please review it again.

File: :w [hunk 4]
```diff
@@ -0,0 +1,353 @@
+package export
```
Comment (kennytm) [inline :w:1]: 🤔

**Pull request closed without merging**